File: backend/app/models/paper.py
"""论文 ORM 模型"""

from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, Boolean
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.utils.database import Base


class Paper(Base):
    __tablename__ = "papers"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(500), nullable=False)
    abstract = Column(Text)
    authors = Column(String(500))
    publication_date = Column(DateTime)
    doi = Column(String(100), unique=True, index=True)
    file_path = Column(String(500))
    file_uuid = Column(String(36), unique=True, nullable=True)
    original_filename = Column(String(255), nullable=True)
    file_size = Column(Integer, nullable=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
    is_favorite = Column(Boolean, default=False)

    user = relationship("User", back_populates="papers")
    ai_analyses = relationship("AIAnalysis", back_populates="paper", cascade="all, delete-orphan")
    tags = relationship("Tag", secondary="paper_tags", back_populates="papers")

    def __repr__(self):
        return f"<Paper(id={self.id}, title={self.title}, user_id={self.user_id})>"

File: frontend/src/types/paper.ts
import type { Tag } from "./tag";

export interface Paper {
  id: number;
  title: string;
  abstract?: string;
  authors?: string;
  publication_date?: string;
  doi?: string;
  is_favorite: boolean;
  file_uuid?: string;
  original_filename?: string;
  file_size?: number;
  tags: Tag[];
  created_at: string;
  updated_at: string;
}

export interface PaperListParams {
  skip?: number;
  limit?: number;
  search?: string;
}

export interface PaperListResponse {
  items: Paper[];
  total: number;
}

export interface BatchActionItem {
  paper_id: number;
  status: "success" | "failed";
  reason: string | null;
}

export interface BatchActionResponse {
  total: number;
  succeeded: number;
  failed: number;
  results: BatchActionItem[];
}
